pathway-analysis/ssgsea_example.Rmd

```diff
@@ -0,0 +1,392 @@
+Again, these were prepared in the 
+[`kegg_ortholog_mapping`](./kegg_ortholog_mapping.nb.html) notebook (see 
+[Gene Sets](#gene-sets) above).
+
+```{r}
+kegg_file <- file.path("gene-sets", 
+                       "c2.cp.kegg.v6.2.entrez_mouse_symbol_hcop.gmt")
+kegg_pathways <- qusage::read.gmt(kegg_file)
+```
+
+### Prep expression matrix
+
+```{r}
+exprs_mat <- symbol_exprs_df %>%
+  tibble::column_to_rownames("SYMBOL") %>%
+  # same order as the metdata -- this may be helpful in downstream analyses
+  dplyr::select(metadata_df$refinebio_accession_code) %>%
+  # qusage takes a matrix
+  as.matrix()
+```
+
+### ssGSEA
+
+We can specify that we'd like to perform ssGSEA with the `method` argument to
+`gsva`.
+
+```{r}
+ssgsea_results <- gsva(expr = exprs_mat,
+                       gset.idx.list = kegg_pathways,
+                       method = "ssgsea",
+                       verbose = FALSE)
+```
+
+This returns a matrix of ssGSEA enrichment scores where the columns are samples
+and the rows are the input gene sets.
+
+```{r}
+dim(ssgsea_results)
+```
+
+```{r}
+head(rownames(ssgsea_results))
+```
+
+#### Identifying "interesting" features
+
+One way to figure out what pathways may be interesting is to find which pathways
+have the highest _variance_ in their ssGSEA scores.
+We can calculate the row variances with the `matrixStats` package.
+
+```{r}
+score_row_var <- matrixStats::rowVars(ssgsea_results)
+# for convenience, let's use the pathway names with this vector
+names(score_row_var) <- rownames(ssgsea_results)
+# look at the 'top' pathways by this metric
+head(sort(score_row_var, decreasing = TRUE))
+```
+
+Note that many of these pathways seem to have something to do with the immune
+system and may very well share a large number of genes.
+Let's look at an example pair of these pathways.
+
+```{r}
+intersect(kegg_pathways$KEGG_ALLOGRAFT_REJECTION, 
+          kegg_pathways$KEGG_GRAFT_VERSUS_HOST_DISEASE)
+```
+
+These ssGSEA scores should not be treated as if they are independent.
+
+#### Plotting features
+
+What do the patterns across strains and tissues look like?
+Let's look at the KEGG Complement and Coagulation Cascades pathway as an 
+example.
+
+We'll be using `ggplot2` for plotting, so we'll want our ssGSEA values as a
+`data.frame`.
+
+```{r}
+comp_coag_df <- data.frame(
+  sample_accession = colnames(ssgsea_results),
+  ssgsea_score = ssgsea_results["KEGG_COMPLEMENT_AND_COAGULATION_CASCADES", ]
+)
+```
+
+We need to add in the strain, tissue, and condition metadata to make an 
+informative plot.
+
+```{r}
+comp_coag_df <- metadata_df %>%
+  dplyr::inner_join(comp_coag_df, 
+                    by = c("refinebio_accession_code" = "sample_accession"))
+```
+
+```{r}
+comp_coag_df %>% 
+  ggplot2::ggplot(ggplot2::aes(x = condition, 
+                               y = ssgsea_score)) +
+  ggplot2::geom_boxplot() +
+  ggplot2::facet_grid(strain ~ refinebio_specimen_part) +
```

Comment:
Is there anyway to adjust the `strain` labels so they are more readable?

Reply:
It would be nice if the graph could be a little taller and less smooshed. But there also might only be so much you can do.

Reply:
TIL You can add `fig.height=11, fig.width=8.5` to the chunk setup ([ref](https://support.rstudio.com/hc/en-us/community/posts/213106048-Figure-size-in-R-Notebook-document))